### PR TITLE
Break down blog-mysql-seed example into packages

### PR DESCRIPTION
**Pull request by victorcolombo**

### CODE REVIEW

File: blog-mysql-seed/blog-seeder/kurtosis.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+name: github.com/kurtosis-tech/blog-post-mysql-seed/blog-seeder/
```
Comment (gbouv) [inline blog-mysql-seed/blog-seeder/kurtosis.yml:1]: nit: newline

File: blog-mysql-seed/mysql/kurtosis.yml [hunk 2]
```diff
@@ -0,0 +1 @@
+name: github.com/kurtosis-tech/blog-post-mysql-seed/mysql/
```
Comment (gbouv) [inline blog-mysql-seed/mysql/kurtosis.yml:1]: nit: newline

File: blog-mysql-seed/mysql/mysql.star [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    )
+
+
+def run_sql(plan, database, sql_query):
+    mysql_flags = "-u {} -p{} -e '{}' {}".format(database.user, database.password, sql_query, database.name)
+    return plan.exec(
+        service_name = database.service.name,
+        recipe = ExecRecipe(command = ["sh", "-c", "mysql {}".format(mysql_flags)]),
+        acceptable_codes = [0, 1, 126]
+    )["output"]
```
Comment (gbouv) [inline blog-mysql-seed/mysql/mysql.star:51]: nit: newline

Comment (gbouv) [inline blog-mysql-seed/mysql/mysql.star:51]: nit: I'd store the result `plan.exec` into a nicely named variable, makes the code cleaner

File: blog-mysql-seed/mysql/mysql.star [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        password=database_password,
+    )
+
+
+def run_sql(plan, database, sql_query):
+    mysql_flags = "-u {} -p{} -e '{}' {}".format(database.user, database.password, sql_query, database.name)
+    return plan.exec(
+        service_name = database.service.name,
+        recipe = ExecRecipe(command = ["sh", "-c", "mysql {}".format(mysql_flags)]),
+        acceptable_codes = [0, 1, 126]
```
Comment (gbouv) [inline blog-mysql-seed/mysql/mysql.star]: Maybe document what each code is here. Really wondering where the 126 comes from and why it's allowed

File: blog-mysql-seed/mysql/mysql.star [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    return struct(
+        service=mysql_service,
+        name=database_name,
+        user=database_user,
+        password=database_password,
+    )
+
+
+def run_sql(plan, database, sql_query):
+    mysql_flags = "-u {} -p{} -e '{}' {}".format(database.user, database.password, sql_query, database.name)
```
Comment (gbouv) [inline blog-mysql-seed/mysql/mysql.star:47]: Wondering what the split is between putting things here VS the `mysql` thing below. Maybe we could add a bit of structure here, like have one string per arg and then concatenating them all together with `mysql ...`

File: blog-mysql-seed/mysql/mysql.star [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    )
+    return struct(
+        service=mysql_service,
+        name=database_name,
+        user=database_user,
+        password=database_password,
+    )
+
+
+def run_sql(plan, database, sql_query):
```
Comment (gbouv) [inline blog-mysql-seed/mysql/mysql.star:46]: No sure how and where this is called, but trusting you on this

File: blog-mysql-seed/mysql/mysql.star [hunk 7]
```diff
@@ -0,0 +1,52 @@
+MYSQL_IMAGE = "mysql:8.0.32"
+
+def create_database(plan, database_name, database_user, database_password, seed_script_artifacts = []):
```
Comment (gbouv) [inline blog-mysql-seed/mysql/mysql.star:3]: And same here, not sure where this is called

File: blog-mysql-seed/mysql/mysql.star [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+    files = {}
+    for index, artifact in enumerate(seed_script_artifacts):
+        files["/docker-entrypoint-initdb.d/{}.sql".format(index)] = artifact
```
Comment (gbouv) [inline blog-mysql-seed/mysql/mysql.star:6]: Can we store this into a constant with a named explaining that this is? I am personally not sure what this does

Comment (gbouv) [reply blog-mysql-seed/mysql/mysql.star:6]: Also, is that expected to have files like `1.sql` `2.sql`, etc? Can we add a prefix that is a little bit descriptive maybe?

File: blog-mysql-seed/mysql/mysql.star [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    service_name = "mysql-{}".format(database_name)
+    mysql_service = plan.add_service(
+        service_name = service_name,
+        config = ServiceConfig(
+            image = MYSQL_IMAGE,
+            ports = {
+                "db": PortSpec(
+                    number = 8080,
```
Comment (gbouv) [inline blog-mysql-seed/mysql/mysql.star:14]: Is that an acceptable default port for mysql? I know postgreas is 5432, 8080 feels kind of to generic for a DB port, but trusting you

File: blog-mysql-seed/mysql/mysql.star [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+                    transport_protocol = "TCP",
+                    application_protocol = "http",
+                ),
+            },
+            files = files,
+            env_vars = {
+                "MYSQL_ROOT_PASSWORD": "root",
```
Comment (gbouv) [inline blog-mysql-seed/mysql/mysql.star:21]: Maybe store `root` in a constant?

File: blog-mysql-seed/mysql/mysql.star [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+                "MYSQL_DATABASE": database_name,
+                "MYSQL_USER": database_user,
+                "MYSQL_PASSWORD":  database_password,
+            },
+        )
+    )
+    # Wait for MySQL to become available
+    mysql_flags = ["-u", database_user, "-p{}".format(database_password), database_name]
```
Comment (gbouv) [inline blog-mysql-seed/mysql/mysql.star:29]: Also wondering why we have such a split here between this and the singleton `["mysql"]` below, feels weird at first glance

**Review by gbouv — APPROVED** [review]
Commented as if I was very new to Starlark. Nothing fully blocking though, feel free to address the comments that feels the most relevant to you and ignoring the rest

File: blog-mysql-seed/mysql/mysql.star (hunk 11 above)
Comment (victorcolombo) [reply blog-mysql-seed/mysql/mysql.star:29]: Done

File: blog-mysql-seed/mysql/mysql.star (hunk 10 above)
Comment (victorcolombo) [reply blog-mysql-seed/mysql/mysql.star:21]: Done

File: blog-mysql-seed/mysql/mysql.star (hunk 8 above)
Comment (victorcolombo) [reply blog-mysql-seed/mysql/mysql.star:6]: they are sorted, so prepending the number guarantees that the order we execute it. Will add a comment to clarify it

File: blog-mysql-seed/mysql/mysql.star (hunk 5 above)
Comment (victorcolombo) [reply blog-mysql-seed/mysql/mysql.star:47]: Done

File: blog-mysql-seed/mysql/mysql.star (hunk 7 above)
Comment (victorcolombo) [reply blog-mysql-seed/mysql/mysql.star:5]: This will be called from another package

File: blog-mysql-seed/mysql/mysql.star (hunk 6 above)
Comment (victorcolombo) [reply blog-mysql-seed/mysql/mysql.star:49]: This will be called from another package

**Pull request merged**